src/main/java/ObjectPool.java

```diff
@@ -0,0 +1,81 @@
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+
+public abstract class ObjectPool<T> implements Pool<T> {
+
+    private int size;
+
+    private boolean shutdown;
+
+    private String host;
+    private int port;
+
+    private BlockingQueue objects;
```

Comment:
BlockingQueue is a typed interface. You don't specify a type nor do you relate the objects contained in the queue to the type parameter specified in the ObjectPool-class (or the implemented interface).

Reconsider what you use this for.

Also: don't call the queue "objects". It is a queue, so call it 'queue' or a specify type of queue.

Reply:
done!